### PR TITLE
chore: remove facades and dart-specific patterns

### PR DESCRIPTION
**Pull request by jelbourn**
R: @hansl @kara @robertmesserle

* Replaces `@OneOf` with string literal types.
* Removes facades and uses standard TS / JS APIs
* Uses standard `angular2/testing` imports

This leaves only two things from facades: `BaseException` and `PromiseCompleter`, which don't have a complete drop-in replacement.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -1,13 +1,10 @@
 import {bootstrap} from 'angular2/platform/browser';
 import {DemoApp} from './demo-app/demo-app';
 import {ROUTER_PROVIDERS} from 'angular2/router';
-import {BrowserDomAdapter} from './core/platform/browser/browser_adapter';
 import {OVERLAY_CONTAINER_TOKEN} from './core/overlay/overlay';
 import {provide} from 'angular2/core';
 import {createOverlayContainer} from './core/overlay/overlay-container';
 
-BrowserDomAdapter.makeCurrent();
```
Comment (hansl) [inline src/main.ts]: This isn't needed anymore?

**Comment by hansl** [comment]
One interrogation, but LGTM!  :shipit:

**Comment by hansl** [comment]
Actually, I'm looking at the errors in Travis and these seem legitimate, based on the change from `OneOf(....)` to string enumeration types. Could you take a deeper look?

**Comment by jelbourn** [comment]
@hansl all better now; it was mostly that `@OneOf` sneakily also set a default value. Also changed sidenav's `toggle` to use a default param and removed a progress-circle test for a behavior it shouldn't actually exhibit.

File: src/main.ts (hunk 1 above)
Comment (jelbourn) [inline src/main.ts]: Nay!

**Comment by jelbourn** [comment]
Also added a type alias in `Dir` to reduce repetition.

File: src/core/annotations/field-value.ts [hunk 2]
```diff
@@ -22,13 +19,13 @@ function booleanFieldValueFactory() {
...
     // Use a fallback if Symbol isn't available.
-    const localKey = isPresent(Symbol) ? Symbol(key) : `__md_private_symbol_${key}`;
+    const localKey = Symbol ? Symbol(key) : `__md_private_symbol_${key}`;
     target[localKey] = defaultValue;
 
     Object.defineProperty(target, key, {
       get() { return this[localKey]; },
       set(value: boolean) {
-        this[localKey] = isPresent(value) && value !== null && String(value) != 'false';
+        this[localKey] = value != null && `${value}` !== 'false';
```
Comment (hansl) [inline src/core/annotations/field-value.ts]: `\`${value}\`` kinda weirds me out. Could you use `toString()` or `'' + ` ? or just `value != 'false'`?

**Comment by hansl** [comment]
Few nits, no blockers. LGTM.

**Comment by hansl** [comment]
Also, this means that unless the template compiler verify typings (which I'm not sure it does), the replacement of `OneOf` by native string literal types may lead to undefined behaviour for typos or invalid values.

File: src/core/annotations/field-value.ts (hunk 2 above)
Comment (jelbourn) [inline src/core/annotations/field-value.ts]: `value != false` isn't allowed by TypeScript (comparing string to boolean).

To me `${value}` is way cleaner than using `toString()` and more readable than `'' + value` (due to the boolean operator) while being the same number of characters).

**Pull request merged**